solutions/src/main/java/ru/sberbank/school/task02/FxClient.java

```diff
@@ -0,0 +1,92 @@
+package ru.sberbank.school.task02;
+
+import lombok.NonNull;
+import ru.sberbank.school.task02.exception.FxConversionException;
+import ru.sberbank.school.task02.util.*;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FxClient implements FxClientController {
+
+    private FxConversionService currentConverter;
+
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss");
+
+
+    FxClient(FxConversionService converter) {
+        currentConverter = converter;
+    }
+
+    @Override
+    public List<FxResponse> fetchResult(@NonNull List<FxRequest> requests) {
+        List<FxResponse> responses = new ArrayList<>();
+        for (FxRequest req : requests) {
+            responses.add(fetchResult(req));
+        }
+        return responses;
+    }
+
+    @Override
+    public FxResponse fetchResult(@NonNull FxRequest requests)
+            throws IllegalArgumentException, NullPointerException {
+        Symbol symbol = getSymbol(requests);
+        BigDecimal amount = getAmount(requests);
+        ClientOperation operation = getOperation(requests);
+
+
+        LocalDateTime localDate = LocalDateTime.now();
+        String date = localDate.format(formatter);
+
+        BigDecimal price = currentConverter.convert(operation, symbol, amount);
+        try {
+            return new FxResponse(symbol.getSymbol(),
+                    price.toString(),
+                    amount.setScale(2, BigDecimal.ROUND_DOWN).toString(),
+                    date,
+                    operation.toString(),
+                    false);
+        }catch (FxConversionException e){
```

Comment:
WhitespaceAround: &#39;}&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a607738515>*